### ISSUE TITLE
Windows Insider Fails to update

### ISSUE THREAD

**Issue by jtsom** [Microsoft/vscode#43830]
### Issue Type
Bug

### Description

Trying to update to lastest Insiders issue fails with errors on Windows.

![image](https://user-images.githubusercontent.com/13488/36309866-36a2a824-12f4-11e8-90ad-c457a6cb1958.png)

From log file:
```
Feb 16 08:30:58.031 INFO Starting: C:\Program Files\Microsoft VS Code Insiders\Code - Insiders.exe, false
Feb 16 08:30:58.041 INFO Checking for running Code - Insiders.exe processes... (attempt 1)
Feb 16 08:30:58.041 INFO Code - Insiders.exe is running, wait a bit
Feb 16 08:30:58.562 INFO Checking for running Code - Insiders.exe processes... (attempt 2)
Feb 16 08:30:58.562 INFO Code - Insiders.exe is not running
Feb 16 08:30:58.571 INFO Starting update, silent = false
Feb 16 08:30:58.701 INFO do_update: "C:\\Program Files\\Microsoft VS Code Insiders\\Code - Insiders.exe", _
Feb 16 08:30:59.131 ERRO FileRec parse error: Failed to parse file rec data

```

### VS Code Info

VS Code version: Code - Insiders 1.21.0-insider (70e020b2377f12894ca11254548ce365a0e435f1, 2018-02-15T05:17:01.988Z)
OS version: Windows_NT x64 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) CPU E5-1620 v2 @ 3.70GHz (2 x 3700)|
|Memory (System)|12.00GB (6.09GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code Insiders\Code - Insiders.exe|
|Screen Reader|no|
|VM|100%|

</details><details><summary>Extensions (15)</summary>

Extension|Author (truncated)|Version
---|---|---
ng-template|Ang|0.1.9
path-intellisense|chr|1.4.2
bracket-pair-colorizer|Coe|0.10.14
vscode-svgviewer|css|1.4.3
githistory|don|0.4.0
auto-rename-tag|for|0.0.15
Angular2|joh|2.12.0
json-to-ts|Mar|1.5.4
markdown-shortcuts|mdi|0.8.1
code-beautifier|mic|1.7.2
prettify-json|moh|0.0.3
debugger-for-chrome|msj|4.1.0
material-icon-theme|PKi|3.2.2
view-in-browser|qin|0.0.5
MDTools|sea|1.0.1

(3 theme extensions excluded)

</details>
Reproduces without extensions

**Comment by joaomoreno** [comment]
@jtsom Can you send me your `C:\Program Files\Microsoft VS Code Insiders\unins000.dat` file?

**Comment by jtsom** [comment]
Still failing to update today. Contents of log file:

Feb 20 08:56:51.776 INFO Starting: C:\Program Files\Microsoft VS Code Insiders\Code - Insiders.exe, false
Feb 20 08:56:51.786 INFO Checking for running Code - Insiders.exe processes... (attempt 1)
Feb 20 08:56:51.786 INFO Code - Insiders.exe is running, wait a bit
Feb 20 08:56:52.296 INFO Checking for running Code - Insiders.exe processes... (attempt 2)
Feb 20 08:56:52.316 INFO Code - Insiders.exe is running, wait a bit
Feb 20 08:56:52.826 INFO Checking for running Code - Insiders.exe processes... (attempt 3)
Feb 20 08:56:52.826 INFO Code - Insiders.exe is running, wait a bit
Feb 20 08:56:53.336 INFO Checking for running Code - Insiders.exe processes... (attempt 4)
Feb 20 08:56:53.336 INFO Code - Insiders.exe is running, wait a bit
Feb 20 08:56:53.846 INFO Checking for running Code - Insiders.exe processes... (attempt 5)
Feb 20 08:56:53.846 INFO Code - Insiders.exe is running, wait a bit
Feb 20 08:56:54.356 INFO Checking for running Code - Insiders.exe processes... (attempt 6)
Feb 20 08:56:54.356 INFO Code - Insiders.exe is not running
Feb 20 08:56:54.366 INFO Starting update, silent = false
Feb 20 08:56:54.496 INFO do_update: "C:\\Program Files\\Microsoft VS Code Insiders\\Code - Insiders.exe", _
Feb 20 08:56:55.186 ERRO FileRec parse error: Failed to parse file rec data


This is after downloading and installing a fresh version last week. This was the first attempt at self updating.

**Comment by jtsom** [comment]
[unins000.dat.zip](https://github.com/Microsoft/vscode/files/1740303/unins000.dat.zip)

**Comment by joaomoreno** [comment]
I can't explain why, but your installer log file seems corrupt. Can you please reinstall VS Code and let me know if you keep getting this issue in the upcoming days?

**Comment by jtsom** [comment]
I will try again - I did reinstall yesterday to get up to date. Yesterday's failure was AFTER I had already reinstalled last week when the issue first came up.

**Comment by joaomoreno** [comment]
What is the unins000.dat size after fully reinstalling?

**Comment by jtsom** [comment]
I just tried to do an update from:

Version 1.21.0-insider
Commit 6309a7734b0b0876b263a6a9cb526fc281100089
Date 2018-02-20T10:24:00.306Z
Shell 1.7.9
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64

I clicked the gear icon, choose Check for Updates. After a little bit, the badge indicated there was an update, so I clicked the gear icon again and choose Install update. Windows asked for permission to do the install and the item in the menu changed to a disabled "installing update". Now the icon has a "1" badge again with "Restart to update". Choosing that I get the windows that are in the original report (installing VS Code with progress bar, and the error dialog). The updater.log file has the same content as previous, and the unins000.dat file is 254,595KB.

I'm going to try a complete uninstall / install (which I've been avoiding, as I don't know if that will wipe out my settings and extensions.)
